snuba/clickhouse/query.py

```diff
@@ -1,8 +1,13 @@
 from typing import NewType
 
+from snuba.query.expressions import Expression as SnubaExpression
 from snuba.query.logical import Query as LogicalQuery
 
 # This defines the type for the Clickhouse query that operates on storages.
 # Obviously this is supposed to become more complex than just a copy of the
 # logical query type.
 Query = NewType("Query", LogicalQuery)
+
+# This defines the type of a Clickhouse query expression. It is used to make
+# the interface of the expression translator more clear.
+Expression = SnubaExpression
```

Comment:
I turned this into a simple type alias because, in order build translation rules that accept very specific types (like Column) instead of a generic expression, such types need to be subtypes of Clickhouse Expression.
```
ClickhouseExpression = NewType("ClickhouseExpression", SnubaExpression)
# Now ClickhosueExpression does not have any subclass. And Column, 
# FunctionCall etc. are not ClickhouseExpression making it not very useful.
```

I don't think this is a problem since, creating this type is only meant to chow what depends on a clickhouse expression in the code and what depends on Snuba Expressions for clarity.

There are two options to consolidate this:
If we built a separate clickhouse AST, they will include different classes, they won't be aliases or NewTypes of each other. Likely they won't even have a common parent class. SnubaExpression and CLickhouseExpression will be siblings at most.

If we decide there is no reason to change the Expression AST (the Query class will definitively be different but in this case the expression AST may be the same data structure) then this alias won't be needed at all. We will just use the same object.